src/bulk/common.ts

```diff
@@ -550,17 +580,21 @@ function mergeBatchResults(batch: any, bulkResult: any, err: any, result: any) {
     }
   }
 
-  if (result.writeConcernError) {
+  if (result?.writeConcernError) {
     bulkResult.writeConcernErrors.push(new WriteConcernError(result.writeConcernError));
   }
 }
 
-function executeCommands(bulkOperation: any, options: any, callback: Callback<BulkWriteResult>) {
+function executeCommands(
+  bulkOperation: BulkOperationBase,
+  options: BulkOptions,
+  callback: Callback<BulkWriteResult>
+) {
   if (bulkOperation.s.batches.length === 0) {
     return callback(undefined, new BulkWriteResult(bulkOperation.s.bulkResult));
   }
 
-  const batch = bulkOperation.s.batches.shift();
+  const batch = bulkOperation.s.batches.shift() as Batch;
 
   function resultHandler(err?: any, result?: any) {
```

Comment:
can we get argument types here?

Reply:
This is where `AnyError` doesn't work because of that darn `.driver` and `.ok` check. I can type `result`. If you have an ideas about `err` I'm all 👂🌽👂.

Reply:
I removed the .driver and cast .ok